File: sdks/php/src/Types/TokenPairResponse.php
<?php

namespace NovaBilling\Types;

use NovaBilling\Core\Json\JsonSerializableType;
use NovaBilling\Core\Json\JsonProperty;

class TokenPairResponse extends JsonSerializableType
{
    /**
     * @var string $accessToken
     */
    #[JsonProperty('accessToken')]
    public string $accessToken;

    /**
     * @var string $refreshToken
     */
    #[JsonProperty('refreshToken')]
    public string $refreshToken;

    /**
     * @param array{
     *   accessToken: string,
     *   refreshToken: string,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->accessToken = $values['accessToken'];
        $this->refreshToken = $values['refreshToken'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}

File: sdks/ruby/lib/novabilling/types/token_pair_response.rb
# frozen_string_literal: true

module Novabilling
  module Types
    class TokenPairResponse < Internal::Types::Model
      field :access_token, -> { String }, optional: false, nullable: false, api_name: "accessToken"
      field :refresh_token, -> { String }, optional: false, nullable: false, api_name: "refreshToken"
    end
  end
end
